src/screens/SendSheet.js

```diff
@@ -302,6 +312,7 @@ export default function SendSheet(props) {
             break;
           default:
             setCurrentNetwork(network);
+            provider = await getProviderForNetwork(networkTypes.mainnet);
```

Comment:
No need for this line since web3Provider is already connected to the app selected network